### PR TITLE
Bug 1978724: Add cypress tests for key/value secrets with binary, ascii, and unicode values.

### PR DESCRIPTION
**Pull request by jcaianirh**
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=1978724

Also addresses: https://issues.redhat.com/browse/CONSOLE-2789

cc: @schituku , @dtaylor113

### CODE REVIEW

File: frontend/packages/integration-tests-cypress/tests/crud/secrets.spec.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    cy.get(infoMessage).should('exist');
+    cy.byTestID('save-changes').click();
+    detailsPage.isLoaded();
+    cy.exec(
+      `oc get secret -n ${testName} ${binarySecretName} --template '{{.data.${secretKey}}}' | base64 -d`,
+      {
+        failOnNonZeroExit: false,
+      },
+    ).then((value) => {
+      cy.fixture('binarysecret.bin', 'binary').then((binarySecret) => {
```
Comment (dtaylor113) [inline frontend/packages/integration-tests-cypress/tests/crud/secrets.spec.ts:55]: Nice use of fixtures!

File: frontend/packages/integration-tests-cypress/tests/crud/secrets.spec.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+  it(`Adds a key/value secret whose value is a unicode file `, () => {
+    cy.get('.co-m-pane__heading').contains('Create key/value secret');
+    cy.byTestID('secret-name').should('exist');
+    cy.byLegacyTestID('file-input-textarea').should('exist');
+    cy.byTestID('secret-name').type(unicodeSecretName);
+    cy.byTestID('secret-key').type(secretKey);
+    cy.byTestID('file-input').attachFile('unicodesecret.utf8');
+    cy.byLegacyTestID('file-input-textarea').should('exist');
+    cy.get(infoMessage).should('not.exist');
+    cy.byTestID('save-changes').click();
```
Comment (dtaylor113) [inline frontend/packages/integration-tests-cypress/tests/crud/secrets.spec.ts:93]: Since this is repeated code, consider making a function `attachFile(secretName, secretKey, file)`

**Review by dtaylor113 — CHANGES_REQUESTED** [review]
Hi @jcaianirh, looks great, only a few small nits. -thanks

File: frontend/packages/integration-tests-cypress/tests/crud/secrets.spec.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    cy.login();
+    cy.visit('/');
+    nav.sidenav.switcher.changePerspectiveTo('Administrator');
+    nav.sidenav.switcher.shouldHaveText('Administrator');
+    cy.createProject(testName);
+  });
+
+  beforeEach(() => {
+    nav.sidenav.clickNavLink(['Workloads', 'Secrets']);
+    cy.byLegacyTestID('resource-title').should('have.text', 'Secrets');
```
[the remark this answers is not in the document]
Comment (jcaianirh) [reply frontend/packages/integration-tests-cypress/tests/crud/secrets.spec.ts:25]: done

File: frontend/packages/integration-tests-cypress/tests/crud/secrets.spec.ts [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    listPage.clickCreateKeyValSecretDropdownButton();
```
[the remark this answers is not in the document]
Comment (jcaianirh) [reply frontend/packages/integration-tests-cypress/tests/crud/secrets.spec.ts:26]: done

**Review by dtaylor113 — APPROVED** [review]
/lgtm
Only concern is perhaps adding a `oc delete secrets` to `after()` to clean up secrets that were created.

**Comment by jcaianirh** [comment]
> /lgtm
> 
> Only concern is perhaps adding a `oc delete secrets` to `after()` to clean up secrets that were created.

Thanks for the review Dave! The secrets are auto deleted when the project is deleted in the last step.

File: frontend/public/components/utils/file-input.tsx [hunk 5]
```diff
@@ -42,11 +42,13 @@ class FileInputWithTranslation extends React.Component<FileInputProps, FileInput
...
     reader.onload = () => {
-      const input = reader.result as string;
+      const input = fileIsBinary
+        ? (reader.result as string).split(',')[1]
+        : (reader.result as string);
       // OnLoad, if inputFileIsBinary we have read as a binary string, skip next block
       if (containsNonPrintableCharacters(input) && !fileIsBinary) {
         fileIsBinary = true;
-        reader.readAsBinaryString(file);
+        reader.readAsDataURL(file);
```
Comment (spadgett) [inline frontend/public/components/utils/file-input.tsx:51]: We'll need a Bugzilla bug for this and backport the fix.

The issue seems to be that we should only use `Base64.encode` on Unicode strings. `btoa` should be used instead of binary data. It's a little strange, but I'm OK using `readAsDataURL` and stripping the prefix.

https://github.com/dankogai/js-base64#decode-vs-atob-and-encode-vs-btoa

**Comment by spadgett** [comment]
/lgtm cancel
Secret tests are failing

**Comment by dtaylor113** [comment]
Hi @jcaianirh , irt the protractor Webhook secrets test failures you are seeing,  locally I'm seeing this at the point of the CI failure:
image
However, in this PR's CI run we see this:
image
Not sure why the webhook secret key is no longer appearing in your PR?

**Comment by jcaianirh** [comment]
> Hi @jcaianirh , irt the protractor Webhook secrets test failures you are seeing, locally I'm seeing this at the point of the CI failure:
> Not sure why the webhook secret key is no longer appearing in your PR?

@dtaylor113 this was causing the webhook tests to fail. I fixed this in my previous commit. If you update to my latest changes, you should no longer see this problem. Thanks!!!

**Comment by spadgett** [comment]
@jcaianirh Can you squash commits?
